test/sockoptapi.js

```diff
@@ -94,6 +94,21 @@ test('sockopt api methods', function(t){
 test('ipv6 socket msg delivery', function (t) {
     t.plan(1);
 
+    var interfaces = require('os').networkInterfaces();
+    for (var iface in interfaces) {
+      var ipv6 = interfaces[iface].filter(function fam(i){
+        return i.family === 'IPv6';
+      });
+      if (ipv6.length) {
+        break;
+      }
+    }
+
+    if (!ipv6.length) {
```

Comment:
I know `ipv6` will get hoisted, but do you mind declaring it earlier & outside of the for loop?